backend/src/test/unimplemented_resources/ID022_collaborationRequestWithMessage.feature

```diff
@@ -0,0 +1,84 @@
+        And the following collaboration request shall exist in the database:
+            | ideaId    | userId    | status  | additionalContact     | message     |
+            | <idea_id> | <user_id> | Pending | "<additionalContact>" | "<message>" |
+        And the number of incoming collaboration requests for the creator of the idea with id "<idea_id>" shall be "<num_incoming_requests>"
+        And the number of outgoing collaboration requests for the user with id "<user_id>" shall be "<num_outgoing_requests>"
+
+        Examples:
+            | email                    | password     | idea_id | user_id | total_num_requests | message                               | additionalContact | num_incoming_requests | num_outgoing_requests |
+            | wassim.jabbour@gmail.com | P@ssWord1234 | 17      | 3       | 5                  | "Hi, please let me join your project" | null              | 3                     | 2                     |
+            | wassim.jabbour@gmail.com | P@ssWord1234 | 18      | 3       | 5                  | "Hi, please let me join your project" | null              | 3                     | 2                     |
+            | john.goblikon@gmail.com  | P@ssWord1234 | 20      | 2       | 5                  | "Hi, please let me join your project" | "428-8794-4237"   | 2                     | 1                     |
+            | john.goblikon@gmail.com  | P@ssWord1234 | 21      | 3       | 5                  | "Hi, please let me join your project" | "428-8794-4238"   | 2                     | 1                     |
+
+
+    # Error flows
+    Scenario Outline: Unsuccessfully send collaboration request
+        Given I am logged in as the user with email "<email>" and password "<password>"
+        When I request to send a collaboration request to the creator of the idea with id "<idea_id>" with message "<message>" and additionalContact "<additionalContact>"
+        Then the number of collaboration requests in the database should be "4"
+        And the number of incoming collaboration requests for the creator of the idea with id "<idea_id>" shall be "<num_incoming_requests>"
+        And the number of outgoing collaboration requests for the user with email "email" shall be "<num_outgoing_requests>"
+        And the following error message shall be returned "<errorMessage>"
+
+        Examples:
+            | email                    | password     | idea_id | message                               | additionalContact | num_incoming_requests | num_outgoing_requests | errorMessage                                                                                              |
+            | jane.doe@gmail.com       | P@ssWord1234 | 17      | "Hi, please let me join your project" | null              | 2                     | 3                     | Cannot send another collaboration requests to the same person before they respond to the previous request |
```

Comment:
Very nice check. Should we make it so that you cannot send another collaboration request to the same person again? Or if the person chooses to ignore, you can only send a certain number more? 

I suggest we do only 1 for now and we can note adding a time lapse or something for a second request as an additional feature